softioc/device.py

```diff
@@ -183,7 +215,14 @@ def _process(self, record):
             self._value = value
             record.UDF = 0
             if self.__on_update and self.__enable_write:
-                dispatcher(self.__on_update, python_value)
+                record.PACT = self._blocking
+                dispatcher(
+                    self.__on_update,
+                    self.__completion,
+                    func_args=(python_value,),
+                    completion_args=(record,)
+                )
```

Comment:
Not a huge fan of the dangling `)` here, FWIW.

Reply:
After all the pain, this has come out remarkably cleanly!

Reply:
Fixed in a0bce0ce330bb96274d4bbebff179c872165d7e6